circuit-benchmarks/src/pi_circuit.rs

```diff
@@ -0,0 +1,115 @@
+
+        let mut rng = XorShiftRng::from_seed([
+            0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0x31, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06,
+            0xbc, 0xe5,
+        ]);
+
+        // Bench setup generation
+        let setup_message = format!("Setup generation with degree = {}", degree);
+        let start1 = start_timer!(|| setup_message);
+        let general_params = ParamsKZG::<Bn256>::setup(degree, &mut rng);
+        let verifier_params: ParamsVerifierKZG<Bn256> = general_params.verifier_params().clone();
+        end_timer!(start1);
+
+        // Initialize the proving key
+        let vk = keygen_vk(&general_params, &circuit).expect("keygen_vk should not fail");
+        let pk = keygen_pk(&general_params, vk, &circuit).expect("keygen_pk should not fail");
+        // Create a proof
+        let mut transcript = Blake2bWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+
+        // Bench proof generation time
+        let proof_message = format!("PI_circuit Proof generation with {} rows", degree);
+        let start2 = start_timer!(|| proof_message);
+        create_proof::<
+            KZGCommitmentScheme<Bn256>,
+            ProverSHPLONK<'_, Bn256>,
+            Challenge255<G1Affine>,
+            XorShiftRng,
+            Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>>,
+            PiCircuit<Fr, MAX_TXS, MAX_CALLDATA>,
+        >(
+            &general_params,
+            &pk,
+            &[circuit],
+            &[&[public_inputs.as_slice()][..]][..],
+            rng,
+            &mut transcript,
+        )
+        .expect("proof generation should not fail");
+        let proof = transcript.finalize();
+        end_timer!(start2);
+
+        // Bench verification time
+        let start3 = start_timer!(|| "PI_circuit Proof verification");
+        let mut verifier_transcript = Blake2bRead::<_, G1Affine, Challenge255<_>>::init(&proof[..]);
+        let strategy = SingleStrategy::new(&general_params);
+
+        verify_proof::<
+            KZGCommitmentScheme<Bn256>,
+            VerifierSHPLONK<'_, Bn256>,
+            Challenge255<G1Affine>,
+            Blake2bRead<&[u8], G1Affine, Challenge255<G1Affine>>,
+            SingleStrategy<'_, Bn256>,
+        >(
+            &verifier_params,
+            pk.get_vk(),
+            strategy,
+            &[&[public_inputs.as_slice()][..]][..],
+            &mut verifier_transcript,
+        )
+        .expect("failed to verify bench circuit");
+        end_timer!(start3);
+    }
+
+    fn fillup_publicdata<const MAX_TXS: usize, const MAX_CALLDATA: usize>() -> PublicData {
+        let mut rng = ChaCha20Rng::seed_from_u64(2);
+
+        let mut public_data = PublicData::default();
+        let chain_id = 1337u64;
+        public_data.extra.chain_id = Word::from(chain_id);
+
+        let n_tx = MAX_TXS;
+        for _ in 0..n_tx {
+            public_data.txs.push(rand_tx(&mut rng, chain_id));
+        }
+
+        public_data
```

Comment:
Maybe `generate_publicdata` is a better name. 

Also this function contains quite a bit of whitespace for a small function. I sometimes do the same thing but I think the other reviewers prefer not to have too much of these, so probably a good idea to remove most of the new lines here.